### PR TITLE
Convert Help and other text-intensive items to external text

### LINKED ISSUE #295: Convert Help and other text-intensive items to external text
The non-climate data pages (Help, etc.) are pretty much entirely text. Such text-intensive pages are hard to maintain and require a full release of the app just to correct a minor error. (Also, these pages are surprisingly unpleasant to code in React (at least in my IDE, they are).)

To the rescue: External text, as developed in/for Plan2Adapt v2. Since it's a small amount of code, but still under development, copy-paste from there to here for now.

### PR DESCRIPTION
**Pull request by rod-glover**
Resolves #295 

The external texts code has had some enhancements and improvements, and is, I believe ready to
be extracted as a separate Nodejs package. I have put in an [issue](https://github.com/pacificclimate/climate-explorer-frontend/issues/304) for this.

To do:

- Help
  - [x] FAQ
  - [x] General
- [x] Science
- [x] About

Guide to reviewers: This PR can be conceptually broken into two parts:
- External text module (`src/utils/external-text.js`)
- Externalization of text content (`src/components/guidance-content/**`, `public/external-text/default.yaml`)

### CODE REVIEW

**Comment by rod-glover** [comment]
This code is ready for review.

File: public/external-text/default.yaml [hunk 1]
```diff
@@ -0,0 +1,1450 @@
...
+              for the data values.</p>
+            </td>
+          </tr>
+          </tbody>
+        </table>
+
+        #### Change from Baseline graph
+
+        See Annual Cycle graph
+
```
Comment (corviday) [inline public/external-text/default.yaml]: Shoot, #299 should have updated this section, but I forgot it existed.

File: public/external-text/default.yaml [hunk 2]
```diff
@@ -0,0 +1,1450 @@
...
+science:
+  title: '# Scientific References'
+  content: |
+    ## Statistically Downscaled Climate Scenarios
+
+    **Bürger, G.**,** T.Q. Murdock**, **A.T. Werner**, **S.R. Sobie**, and **A.J. Cannon**, 2012:
+    [Downscaling extremes - an intercomparison of multiple statistical methods for present climate](http://journals.ametsoc.org/doi/abs/10.1175/JCLI-D-11-00408.1).
+    _Journal of Climate_, **25**, 4366–4388. doi:10.1175/JCLI-D-11-00408.1.
+
+    **Bürger, G.**, **S.R. Sobie**, **A.J. Cannon**, **A.T. Werner**, and **T.Q. Murdock**, 2013:
```
Comment (corviday) [inline public/external-text/default.yaml]: This paper is repeated twice.

File: public/external-text/default.yaml [hunk 3]
```diff
@@ -0,0 +1,1450 @@
...
+    items:
+      - header: Feedback on Application
+        body: |
+          Please address questions and suggestions on the functioning of this
+          tool (the application proper) to
+          [climate@uvic.ca](mailto:climate@uvic.ca).
+      - header: Scientific Questions
+        body: |
+          Please address questions about science and interpretation of the
+          data presented in this tool to [TBD](#).
```
Comment (corviday) [inline public/external-text/default.yaml]: Can this easily be un-linkified? Clicking on it is possible, and takes you back to the data. Not really a serious issue - who's gonna click on it? - but odd.

File: src/utils/external-text.js [hunk 4]
```diff
@@ -0,0 +1,299 @@
+// External text module.
```
Comment (corviday) [inline src/utils/external-text.js]: Very nice how documented this is!

File: public/external-text/default.yaml (hunk 1 above)
Comment (corviday) [reply public/external-text/default.yaml]: https://github.com/pacificclimate/climate-explorer-frontend/issues/305

File: public/external-text/default.yaml [hunk 5]
```diff
@@ -0,0 +1,1450 @@
...
+  
+# Bits and chunks used repeatedly in other parts of this document.
+# (Referred to in other elements via `${$$.components}`.)
+# For sanity, let's keep these in alphabetical order.
+components:
+  cddCaution: |
+    Please note the distinction between the variable we label `cdd`,
+    meaning cooling degree-days, and the Climdex variable `CDD`,
+    meaning maximum length of dry spell, which we label `cddETCCDI`.
+  climdexUrl: https://www.climdex.org/learn/indices/
```
[the remark this answers is not in the document]
Comment (rod-glover) [reply public/external-text/default.yaml]: As far as I understand it, our implementation is supposed to follow these definitions. They're the standard, and we shouldn't be calling them Climdex indices if they are different. That said, I'll ask Trevor. For now I will assume this is correct.

**Comment by rod-glover** [comment]
Links fixed. I'll leave #305 to you for after this branch is merged.

**Comment by rod-glover** [comment]
As to instructions about adding a new variable, that belongs in README, not in the user-facing Help.

**Comment by corviday** [comment]
As I understand it, part of the changes in your last two PRs were to make it possible to update the variable configuration file and variable descriptions in the help (which one might like to update when a new variable is added) without redeploying the front end. I am not sure how that process would work.

File: public/external-text/default.yaml [hunk 6]
```diff
@@ -0,0 +1,1450 @@
...
+
+          For details, see [Climdex variable `TN90P`](${$$.components.climdexUrl}/#index-TN90P).
+
+        * **`tnnETCCDI`** 
+          
+          Annual minimum of daily minimum temperature
+
+          For details, see [Climdex variable `TNN`](${$$.components.climdexUrl}/#index-TNN).
+
+        * **`tnnETCCDI`**
```
Comment (corviday) [inline public/external-text/default.yaml]: Do the two tnnETCCDI climdices need the monthly / annul ambiguity boilerplate?

File: public/external-text/default.yaml [hunk 7]
```diff
@@ -0,0 +1,1450 @@
...
+
+          For details, see [Climdex variable `TNN`](${$$.components.climdexUrl}/#index-TNN).
+
+        * **`tnxETCCDI`** 
+          
+          Annual maximum of daily minimum temperature
+
+          For details, see [Climdex variable `TNX`](${$$.components.climdexUrl}/#index-TNX).
+
+        * **`tnxETCCDI`**
```
Comment (corviday) [inline public/external-text/default.yaml]: Do the tnxETCCDI climdices need the monthly/annual warning?

**Comment by rod-glover** [comment]
Well spotted, @corviday . All fixed now. Also some broken links fixed. Going to merge now.

**Pull request merged**